pkg/config/config_template.yaml

```diff
@@ -1141,13 +1141,13 @@ api_key:
 
   ## @param filter_tags - object - optional
   ## Defines rules by which to filter traces based on tags.
-  ##  * require - list of key or key/value strings - traces must have those tags in order to be sent to Datadog
-  ##  * reject - list of key or key/value strings - traces with these tags are dropped by the Agent
+  ##  * require - list of key or key/value regex pattern strings - traces must have those tags in order to be sent to Datadog
+  ##  * reject - list of key or key/value regex pattern strings - traces with these tags are dropped by the Agent
   ## Note: Rules take into account the intersection of tags defined.
   #
   # filter_tags:
-  #     require: [<LIST_OF_KEY_VALUE_TAGS>]
-  #     reject: [<LIST_OF_KEY_VALUE_TAGS>]
```

Comment:
Changing explicit matching to regex matching is a breaking change.  For example, `http.path_group` will now match more strings (such as `http_path_group`).

One option here is to use a different JSON structure for the two

```yaml
require: ['foo.bar', {"regex": "foo\..*\.bar"}]
```
Here, the new syntax is never valid for older agents, so there's no conflict. However, parsing this with the existing config library may be difficult.

Reply:
I'll try to think how to avoid this breaking change soon

Reply:
Maybe the right solution would be to add a new block(filter_tags_regex) to the configuration and corresponding env variables for reject and require?

Reply:
That could make sense.  You'd need to be careful to define the interactions: spans that match (require OR require_regex) AND NOT (reject or reject_regex)

Reply:
I add new configuration option `filter_tags_regex`.

When `filter_tags` and `filter_tags_regex` are used at the same time, all rules are united for filtering.
In the case when rule in `filter_tags` and `filter_tags_regex` have the same key, rule from `filter_tags` have precendence over the `filter_tags_regex`.